tests/tests.go

```diff
@@ -359,8 +376,70 @@ func (ctl *controller) approveIssue(issue *api.Issue) error {
 
 					return fmt.Errorf("failed to patch task status for task %d, error: %w", task.ID, err)
 				}
+				return nil
 			}
 		}
 	}
 	return nil
 }
+
+type pipelineStatus string
```

Comment:
I am wondering if we need to introduce the extra pipeline state. What's the limitation if we just use the task state?

Reply:
Done. Was trying to show the overall pipeline status. Changed method name to getAggregatedTaskStatus() so that the name reflects its usage.